ndk/src/main/jni/metadata.c

```diff
@@ -145,6 +145,25 @@ int bsg_get_map_value_int(JNIEnv *env, bsg_jni_cache *jni_cache, jobject map,
   return 0;
 }
 
+int bsg_populate_api_level_from_map(JNIEnv *env, bsg_jni_cache *jni_cache, jobject device_data) {
+    int value = -1;
+    jstring map_key = (*env)->NewStringUTF(env, "runtimeVersions");
+    jstring version_key = (*env)->NewStringUTF(env, "androidApiLevel");
+    jobject _map = (*env)->CallObjectMethod(env, device_data, jni_cache->hash_map_get, map_key);
+
+    if (_map != NULL) {
+        jobject _versions =
```

Comment:
`_versions` could have a more appropriate name here – it's the value of "androidApiLevel", so I guess it could just be `_apiLevel` or something.

Reply:
I've renamed the variables in this method so they're a little less confusing.